SCPCP000006_05_explore_COPYKAT (#813)
AlexsLemonade/OpenScPCA-analysis: analyses/cell-type-wilms-tumor-06/notebook_template/05_copykat_exploration.Rmd

date: '2024-08-13'
params:
  sample_id: "SCPCS000194"
  seed: 12345
output: 
  html_document: 
    toc: yes
    toc_float: yes
    code_folding: hide
    highlight: pygments
    df_print: paged
---

```{r setup, include=FALSE}

knitr::opts_chunk$set(echo = TRUE,
                      message=FALSE,
                      warnings=FALSE
)

```

## Introduction

```{r}

subdiagnosis <- readr::read_tsv(
  file.path("..", "..", "..", "data", "current", "SCPCP000006", "single_cell_metadata.tsv"),
  show_col_types = FALSE
) |>
  dplyr::filter(scpca_sample_id == params$sample_id) |>
  dplyr::pull(subdiagnosis)

```

This notebook explores using [`CopyKAT`](https://github.com/navinlabcode/copykat) to estimate tumor and normal cells in `r params$sample_id` from SCPCP000006. 
This sample has a(n) `r subdiagnosis` subdiagnosis.

`CopyKAT` was run using the `copyKAT.R` script with and without a normal reference, using either an euclidean or statistical (spearman) method to calculate distance in `copyKAT`. 

These results are read into this notebook and used to: 
  
  - Visualize diploid and aneuploid cells on the UMAP. 
  - Evaluate common copy number gains and losses in Wilms tumor. 

>   - Calculate the confusion matrix comparing manual annotations of tumor cells to using `CopyKAT` to annotate tumor cells. 

Reviewer: You didn't adapt this part of the original notebook:
```suggestion
```

full_cnv_df <- full_ck_results_df |>
    tidyr::pivot_longer(
    cols = -c(
    reference_used,
    chrom
    ),
    names_to = "barcodes",
    values_to = "cnv_detection"
    ) |>
    dplyr::group_by(chrom, barcodes, reference_used) |>
    dplyr::summarise(mean_cnv_detection = mean(cnv_detection))
                                                                                                        
# join with cnv info
cnv_df <- cnv_df |>
   dplyr::left_join(full_cnv_df, by = c("barcodes", "reference_used")) |>
   dplyr::filter(!is.na(chrom))    

```
                                                                                                        
Let's look at the distribution of CNV estimation in cells that are called aneuploid and diploid by `CopyKAT`. 

```{r, fig.height=15, fig.width=10}
# create faceted density plots showing estimation of CNV detection across each chr of interest
# colored by aneuploid/diploid estimation
ggplot(cnv_df, aes(x = mean_cnv_detection, color = copykat.pred)) +
  geom_density() +
  theme_bw() +
  facet_grid(
    rows = vars(chrom),
    cols = vars(reference_used)
  )
```

## Conclusions

> From the heatmap of CNV and the mean CNV detection plots, it is difficult to see any CNV pattern that drive the identification of aneuploid cells. 

Reviewer: It's a bit worse than you describe...!

```suggestion
From the heatmap of CNV and the mean CNV detection plots, there does not appear to be any pattern that drives the identification of aneuploid cells. 
```

The assignment of the aneuploidy/diploidy value might relies on very few CNV and/or an arbitrary threshold. 

> This might be why the assignement of aneuploidy/diploidy values differs between condition (and between runs!!).

Author: yes, I figured this out when running multiple times to save the heatmap not in jpeg but png format. I can send you examples tomorrow (I struggle with the vpn connexion now 😞, hopefully better tomorrow! )

> This might be why the assignement of aneuploidy/diploidy values differs between condition (and between runs!!).

Author: @sjspielman some example to illustrate. 
For the sample -208, I show you the heatmap of CNV for `copykat` ran with an euclidean distance, no reference, two days apart (to then save the heatmap in png instead of jpeg:
![image](https://github.com/user-attachments/assets/7f419505-42e1-4f1b-8f5a-8162e13bfec5)
![image](https://github.com/user-attachments/assets/9a879e3a-5260-43f1-ba16-a72da0839da9)

This is not completly different but not identical.

> This might be why the assignement of aneuploidy/diploidy values differs between condition (and between runs!!).

Author: for sample -205
![image](https://github.com/user-attachments/assets/ea77c0f9-7097-47f3-a2cd-94f70407164c)
![image](https://github.com/user-attachments/assets/a18ceb88-a3b9-456c-bd23-7bee8c4f15a1)

> This might be why the assignement of aneuploidy/diploidy values differs between condition (and between runs!!).

Author: for -184 it is much more striking ❓❗ 
![image](https://github.com/user-attachments/assets/eee3db63-2145-4a9e-92b8-674c9b7f1157)
